packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/LayoutPropertyApplicatorTest.kt

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.uimanager
+
+import org.mockito.Mockito.never
+import org.mockito.Mockito.reset
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import android.util.DisplayMetrics
+import com.facebook.react.bridge.JavaOnlyMap
+import com.facebook.react.uimanager.Spacing.*
+import com.facebook.yoga.YogaAlign
+import com.facebook.yoga.YogaConstants
+import com.facebook.yoga.YogaFlexDirection
+import com.facebook.yoga.YogaJustify
+import com.facebook.yoga.YogaPositionType
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.Assert.*
+import org.junit.Ignore
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.eq
+import org.mockito.ArgumentMatchers.anyFloat
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.Mockito.mockStatic
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+@Ignore
```

Comment:
Have you tried to remove this `@Ignore` and fix the tests?

Reply:
Yes I have; I only added the `@Ignore` on [0ba9f2f](https://github.com/facebook/react-native/pull/41649/commits/0ba9f2fa67f9043ab2c1a384c916e7cdca267c17) to follow the approach suggested on #39284 
The issues I'm facing are related to the issues reported on #38825, where Yoga is not mockable, which is a dependency for `LayoutShadowNode`. I tried using the ShadowSOLoader as you suggested on the thread, but it still didn't fix the issue, just changed the error stack to: 
```
Could not initialize class com.facebook.react.uimanager.LayoutShadowNode
java.lang.NoClassDefFoundError: Could not initialize class com.facebook.react.uimanager.LayoutShadowNode
	at com.facebook.react.uimanager.LayoutPropertyApplicatorTest.testDimensions(LayoutPropertyApplicatorTest.kt:57)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:580)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$2(SandboxTestRunner.java:287)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:99)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: 'long com.facebook.yoga.YogaNative.jni_YGConfigNewJNI()' [in thread "SDK 33 Main Thread"]
	at com.facebook.yoga.YogaNative.jni_YGConfigNewJNI(Native Method)
	at com.facebook.yoga.YogaConfigJNIBase.<init>(YogaConfigJNIBase.java:23)
	at com.facebook.yoga.YogaConfigJNIFinalizer.<init>(YogaConfigJNIFinalizer.java:12)
	at com.facebook.yoga.YogaConfigFactory.create(YogaConfigFactory.java:12)
	at com.facebook.react.uimanager.ReactYogaConfigProvider.get(ReactYogaConfigProvider.java:20)
	at com.facebook.react.uimanager.ReactShadowNodeImpl.<clinit>(ReactShadowNodeImpl.java:62)
	at com.facebook.react.uimanager.LayoutPropertyApplicatorTest.testPadding(LayoutPropertyApplicatorTest.kt:195)
```

Reply:
Without the SoLoader as a config for the test class, the error stack is:
```
no yoga in java.library.path: /Users/marcelcanhisares/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:.
java.lang.UnsatisfiedLinkError: no yoga in java.library.path: /Users/marcelcanhisares/Library/Java/Extensions:/Library/Java/Extensions:/Network/Library/Java/Extensions:/System/Library/Java/Extensions:/usr/lib/java:.
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2434)
	at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
	at java.base/java.lang.System.loadLibrary(System.java:1989)
	at com.facebook.soloader.SoLoader.loadLibraryOnNonAndroid(SoLoader.java:926)
	at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:882)
	at com.facebook.soloader.SoLoader.loadLibrary(SoLoader.java:869)
	at com.facebook.yoga.YogaNative.<clinit>(YogaNative.java:16)
	at com.facebook.yoga.YogaConfigJNIBase.<init>(YogaConfigJNIBase.java:23)
	at com.facebook.yoga.YogaConfigJNIFinalizer.<init>(YogaConfigJNIFinalizer.java:12)
	at com.facebook.yoga.YogaConfigFactory.create(YogaConfigFactory.java:12)
	at com.facebook.react.uimanager.ReactYogaConfigProvider.get(ReactYogaConfigProvider.java:20)
	at com.facebook.react.uimanager.ReactShadowNodeImpl.<clinit>(ReactShadowNodeImpl.java:62)
	at com.facebook.react.uimanager.LayoutPropertyApplicatorTest.testDimensions(LayoutPropertyApplicatorTest.kt:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:580)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$2(SandboxTestRunner.java:287)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:99)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)

```

Reply:
> Yes I have; I only added the @Ignore on [0ba9f2f](https://github.com/facebook/react-native/pull/41649/commits/0ba9f2fa67f9043ab2c1a384c916e7cdca267c17) to follow the approach suggested on https://github.com/facebook/react-native/pull/39284

Ah I see. Ok let's merge it as is and I'll do a pass afterwards in trying to fix this and other suppressed tests.